src/app/clusters/user-label-server/user-label-server.cpp

```diff
@@ -74,7 +78,27 @@ CHIP_ERROR UserLabelAttrAccess::ReadLabelList(EndpointId endpoint, AttributeValu
     return err;
 }
 
-UserLabelAttrAccess gAttrAccess;
+CHIP_ERROR UserLabelAttrAccess::WriteLabelList(EndpointId endpoint, AttributeValueDecoder & aDecoder)
+{
+    DeviceLayer::LabelList<app::Clusters::UserLabel::Structs::LabelStruct::Type, DeviceLayer::kMaxUserLabels> labelList;
+    LabelList::TypeInfo::DecodableType decodablelist;
+
+    ReturnErrorOnFailure(aDecoder.Decode(decodablelist));
+
+    uint8_t index = 0;
+    auto iter     = decodablelist.begin();
+    while (iter.Next())
+    {
+        auto & entry = iter.GetValue();
+
+        VerifyOrReturnError(index < DeviceLayer::kMaxUserLabels, CHIP_ERROR_BUFFER_TOO_SMALL);
```

Comment:
Why do we need a separate `index`?  Why can't we use `labelList.size()`?   Or just use the fact that `add()` returns error on out of space and is already doing this length check anyway?

Reply:
Acked